app/validators/blocks/calculated_summary_block_validator.py

```diff
@@ -19,13 +20,34 @@ def validate(self):
         if (answers := self.get_answers(self.answers_to_calculate)) is None:
             return self.errors
 
+        self.validate_single_answer_is_for_dynamic_answers(answers)
         self.validate_answer_id_set_before_calculated_summary_block()
         self.validate_answer_id_for_calculated_summary_not_in_different_section()
 
         self.validate_answer_types(answers)
 
         return self.errors
 
+    def validate_single_answer_is_for_dynamic_answers(self, answers: list[dict]):
+        """Validate that if there is only one answer in the answers_to_calculate list, it's for dynamic answers"""
+        if len(answers) > 1:
+            return
+        # minItems = 1 so answers won't be empty
+        single_answer_id = answers[0]["id"]
+        question_block_id = self.questionnaire_schema.get_block_id_by_answer_id(
+            single_answer_id
+        )
+        if (
+            single_answer_id
+            not in self.questionnaire_schema.get_all_dynamic_answer_ids(
+                question_block_id
+            )
+        ):
```

Comment:
```suggestion
        if len(answers) == 1 and answers[
                0
            ] not in self.questionnaire_schema.get_all_dynamic_answer_ids(
                self.block["id"]
            ):
```

I would make it more compacted, especially when you have self.block unlike me in my PR, docstring explains it already. I don't know what others think?

Reply:
Yeah good point - can definitely make it shorter, though I just realised that using `self.block["id"]` isn't working as that is the block id of the calculated summary rather than the block id of the question for the single answer, I think its going to be the same problem on your PR

Reply:
I kept `single_answer_id` as its used twice and I think it helps with readability, but made the rest more compact as per suggestion!